### PR TITLE
Move limiter out to a separate package and add tests

### PR DESCRIPTION
**Pull request by prashantv**
Prefactor out the limiter into a separate package from #58

### CODE REVIEW

File: limiter/limiter.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	r := &Run{
+		requestsLeft: int64(maxRequests),
+		limiter:      limiter,
+	}
+	time.AfterFunc(maxDuration, func() {
+		r.Stop()
+	})
+
```
Comment (abhinav) [inline limiter/limiter.go]: Nit:

    time.AfterFunc(maxDuration, r.Stop)

File: limiter/limiter.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	return r
+}
+
+// More returns whether more requests can be started or whether we have
+// reached some limit.
+func (r *Run) More() bool {
```
Comment (abhinav) [inline limiter/limiter.go]: Mention that this blocks until the request can be sent.

**Comment by abhinav** [comment]
LGTM with nits

File: limiter/limiter.go (hunk 1 above)
Comment (prashantv) [inline limiter/limiter.go]: 👍

File: limiter/limiter.go (hunk 2 above)
Comment (prashantv) [inline limiter/limiter.go]: 👍 
Also updated `More` so it doesn't `Take` if `requestsLeft` is already 0

**Pull request merged**